### DISCUSSION TITLE
Email Configuration

### DISCUSSION THREAD

**Discussion by StefanQuick** [docmost/docmost#339] [Help / Questions]
"To send emails, you will have to configure the email driver."

Well how do i find the email driver?

**Comment by Philipinho** [comment]
You have to fill-in the SMTP environment variables found here https://docmost.com/docs/self-hosting/configuration/#using-smtp

You can use any SMTP provider. Examples are mailgun, postmark and sendgrid.

**Reply by StefanQuick** [reply]
Well you did not awnser my question, Where is it?

I have an smtp server and all variables, but where do i set the:

AIL_DRIVER=smtp
SMTP_HOST=
SMTP_PORT=
SMTP_USERNAME=
SMTP_PASSWORD=
SMTP_SECURE=
MAIL_FROM_ADDRESS=hello@example.com
MAIL_FROM_NAME=Docmost 

Which file do i write all that?

**Reply by StefanQuick** [reply]
I searched the web like a maniac yesterday and found out you should probably write it in the docker-compoese.yml and wrote this below, but still does not work!!! Am i in the wrong file??? In your documentation it says i should use equal sign but this file only takes colons. So am i in the wrong file our does your documentation fail?

services:
  docmost:
    image: docmost/docmost:latest
    depends_on:
      - db
      - redis
    environment:
      APP_URL: 'http://docmost.xxx.se:3000'
      APP_SECRET: 'xxx'
      DATABASE_URL: 'xxx'
      REDIS_URL: 'redis://redis:6379'
      MAIL_DRIVER: 'smtp'
      SMTP_HOST: 'mail.xxx.se'
      SMTP_PORT: 25
      SMTP_USERNAME: 'docmost'
      SMTP_PASSWORD: 'docmost123'
      SMTP_SECURE: 'false'
      MAIL_FROM_ADDRESS: 'docmost@xxx.se'
      MAIL_FROM_NAME: 'Docmost'
    ports:
      - "3000:3000"
    restart: unless-stopped
    volumes:
      - docmost:/app/data/storage

**Reply by Philipinho** [reply]
Here is an example.
Make sure the environment variables are properly indented.
```dockerfile
version: "3"

services:
  docmost:
    image: docmost/docmost:latest
    depends_on:
      - db
      - redis
    environment:
      APP_URL: "http://localhost:3000"
      APP_SECRET: "REPLACE_WITH_LONG_SECRET"
      DATABASE_URL: "postgresql://docmost:STRONG_DB_PASSWORD@db:5432/docmost?schema=public"
      REDIS_URL: "redis://redis:6379"
      MAIL_DRIVER: “smtp”
      SMTP_HOST: “mail.example.com”
      SMTP_PASSWORD: “PASWORD_HERE”
      SMTP_PORT: 25
      …other_mail_configurations
    ports:
      - "3000:3000"
    restart: unless-stopped
    volumes:
      - docmost:/app/data/storage

  db:
    image: postgres:16-alpine
    environment:
      POSTGRES_DB: docmost
      POSTGRES_USER: docmost
      POSTGRES_PASSWORD: STRONG_DB_PASSWORD
    restart: unless-stopped
    volumes:
      - db_data:/var/lib/postgresql/data

  redis:
    image: redis:7.2-alpine
    restart: unless-stopped
    volumes:
      - redis_data:/data

volumes:
  docmost:
  db_data:
  redis_data:

```

**Comment by StefanQuick** [comment]
ERROR [EmailProcessor] Error processing send-email job. Reason: unable to get local issuer certificate

I have tryed every combination of secure, ignoretls and port, without success.
I have tryed alot of combinations of SMTP_HOST and SMTP_TLSSERVERNAME: with ip and dns names.
 
 I have looked on https://nodemailer.com/smtp/ and tryed a few parameters, but it seems not to have implemted them all? 
 
 We use a Exchange2019 with TLS1.2 that dont seems to work.

**Reply by Philipinho** [reply]
Not all nodemailer options are available.  
`ignoreTLS` option will be available in the next version. It might help in this case.
